custom_metrics/third-parties.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Detect the presence of third party libraries.
+ *
+ * Look for common library aliases on the global scope.
+ * If available, detect libraries' versions.
+ *
+ * Outputs JSON-serialized list of library-version pairs.
+ *     e.g.: `["a@1.0","b","c@2.2"]`
+ */
+var thirdParties = [];
+var name, version;
+
+function addThirdParty(name, version) {
+	if (!name) {
+		return;
+	}
+
+	if (version) {
+		name += '@' + version;
```

Comment:
I think it would also make DB queries easier if the version part was always appended (even if unknown/null), so that there's no need to distinguish in the query between the two cases "name" and "name@version", because it would either be "name@1.2.3" or "name@unknown".